### DISCUSSION TITLE
Is it compatible with VueJS 2.7?

### DISCUSSION THREAD

**Discussion by MartinX3** [vuetifyjs/vuetify#15367] [Q&A] [answered]
Or do we need to wait for VuetifyJS 2.7?

https://github.com/vuejs/vue/releases/tag/v2.7.0

**Comment by KaelWD** [comment, marked as answer]
Vuetify itself is, vuetify-loader isn't yet: https://github.com/vuetifyjs/vuetify-loader/issues/240

**Reply by MartinX3** [reply]
@KaelWD 
Thank you for your answer
Can we just force 2.7 ourselves or should we wait for the next vuetify 2.6 patch release?

**Reply by KaelWD** [reply]
You just need to update vuetify-loader to [v1.9.0](https://github.com/vuetifyjs/vuetify-loader/releases/tag/v1.9.0)

**Reply by MartinX3** [reply]
Ah, thank you.
I asked because you also changed code with your latest vuetify commit, @KaelWD

**Reply by KaelWD** [reply]
Shouldn't affect consumers, it was all typescript/internal testing stuff.

**Reply by MartinX3** [reply]
Ah, that's great to know.
If it doesn't affect my using typescript as well, it's fine.
Thank you very much!

**Reply by dparish** [reply]
What about Vue Cli 5 so we can use web pack 5?

I tried it and it appears as if the components are not auto loaded.

**Reply by MartinX3** [reply]
Same for vite.
It's only usable with heavy workarounds.
I assume we need to wait for vuetify 2.7

**Reply by kingyue737** [reply]
Hey @MartinX3, I've made a [starter template](https://github.com/kingyue737/vitify-admin) with Vite + Vuetify 2 +Vue 2.7, components are auto imported by [unplugin-vue-components](https://github.com/antfu/unplugin-vue-components). It works like a charm. Several products of my team based on this template are in stable production. Hope it can help you.

**Reply by MartinX3** [reply]
@kingyue737 thank you but sadly it seems that there is too much code to workaround the missing vite support or it takes too much time to analyse the project and isolate and extract the specific code to add vite support.
I better wait for vuetify 2.7/3.2

**Reply by kingyue737** [reply]
Sorry the project is a bit heavy for beginer, the docs is still in progress. It's not responsibility of vuetify 2.7 to support vite but an official `vite-plugin-vuetify2` which doesn't exist currently. In fact, it's very simple to use vuetify 2 in vite!! Just intall [unplugin-vue-components](https://github.com/antfu/unplugin-vue-components).

```
npm install -D unplugin-vue-components
```
then add the plugin in `vite.config.ts`
```ts
import { defineConfig } from 'vite'
import vue2 from '@vitejs/plugin-vue2'
import Components from 'unplugin-vue-components/vite'
import { VuetifyResolver } from 'unplugin-vue-components/resolvers'

defineComponent({
  plugins: [
     vue2(),
     Components({
        resolvers: [VuetifyResolver()],
     })
  ]
})
```
Not `too much` at all.

If you also use vuetify [custom sass variable](https://vuetifyjs.com/en/features/sass-variables/)
```ts
defineComponent({
  plugins: [
    vue2(),
    Components({
      resolvers: [VuetifyResolver()],
    }),
  ],
  css: {
    // https://vitejs.dev/config/#css-preprocessoroptions
    preprocessorOptions: {
      sass: {
        additionalData: [
          // vuetify variable overrides
          '@import "@/styles/variables.scss"',
          '',
        ].join('\n'),
      },
    },
  },
})
```
9 extra lines in vite config. That's all.

If you don't want to wait, take a try. And don't hesitate to ask in discussion if you meet any issue

**Reply by MartinX3** [reply]
Thank you, I keep it in mind to test it later with `Nuxt 2 Bridge` which I use as static website generator.

Edit: Sadly not working.
